sw/applications/spi_flash_write/spi_flash_write.c

```diff
@@ -0,0 +1,274 @@
+// Copyright EPFL contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "core_v_mini_mcu.h"
+#include "csr.h"
+#include "hart.h"
+#include "handler.h"
+#include "rv_plic.h"
+#include "rv_plic_regs.h"
+#include "soc_ctrl.h"
+#include "spi_host.h"
+#include "dma.h"
+#include "fast_intr_ctrl.h"
+#include "fast_intr_ctrl_regs.h"
+
+// Un-comment this line to use the SPI FLASH instead of the default SPI
+#define USE_SPI_FLASH
+
+#define COPY_DATA_BYTES 256 // Flash page size = 256
+#define SPI_BYTES (4 * (uint32_t)((COPY_DATA_BYTES-1) / 4 + 1)) // Only sends data when an entire word has been received
+
+#define FLASH_ADDR 0x00008500 // 256B data alignment
+
+int8_t spi_intr_flag;
+spi_host_t spi_host;
+
+void handler_irq_fast_spi(void)
+{
+    // Disable SPI interrupts
+    spi_enable_evt_intr(&spi_host, false);
+    spi_enable_rxwm_intr(&spi_host, false);
+    // Clear fast interrupt
+    fast_intr_ctrl_t fast_intr_ctrl;
+    fast_intr_ctrl.base_addr = mmio_region_from_addr((uintptr_t)FAST_INTR_CTRL_START_ADDRESS);
+    clear_fast_interrupt(&fast_intr_ctrl, kSpi_e);
+    spi_intr_flag = 1;
+}
+
+void handler_irq_fast_spi_flash(void)
+{
+    // Disable SPI interrupts
+    spi_enable_evt_intr(&spi_host, false);
+    spi_enable_rxwm_intr(&spi_host, false);
+    // Clear fast interrupt
+    fast_intr_ctrl_t fast_intr_ctrl;
+    fast_intr_ctrl.base_addr = mmio_region_from_addr((uintptr_t)FAST_INTR_CTRL_START_ADDRESS);
+    clear_fast_interrupt(&fast_intr_ctrl, kSpiFlash);
+    spi_intr_flag = 1;
+}
```

Comment:
Can you but them in an ifndef with USE_SPI_FLASH for a better understanding (except if both are actually needed in both cases)